### ISSUE TITLE
Question about .pb model file

### ISSUE THREAD

**Issue by FantasyJXF** [LouieYang/stroke-controllable-fast-style-transfer#4]
Hi, I visualize your pretrained .pb model with Netron, and find that the `content_input` tensor shaped like `(1, None, None, 3)`

<img width="640" alt="111" src="https://user-images.githubusercontent.com/16235526/59733098-49501e00-927f-11e9-96f3-dfe3dce16643.png">

But I checked your code, the shape is (batch_size, None, None, 3), and the default batch size is 2, so why  the `content_input` tensor in your  pretrained model is (1, None, None, 3)?

I'm not sure whether you trained with batch_size=1, that might be very slow?

```
self.content_input = tf.placeholder(tf.float32, shape=(args.batch_size, None, None, 3), name='content_input')
```

Looking forward to your reply.

**Comment by ycjing** [comment]
Hi @FantasyJXF  Setting the width and height as "None" allows arbitrary input sizes.

**Comment by ycjing** [comment]
For the batch size, we experimentally find that  setting batch size as 1 leads to more stable results.

**Comment by FantasyJXF** [comment]
OK, thanks

**Comment by FantasyJXF** [comment]
激动的我想说中文，你真的是太优秀了，得有多熟练才能随意的把fast-style-transfer的代码一顿魔改，让自己的算法得以实现。

看了许多你关于图像纹理方向的论文，感谢贡献！

**Comment by ycjing** [comment]
谢谢~ 其实是我师弟 @LouieYang 给力 :+1:

**Comment by FantasyJXF** [comment]
![0_5173](https://user-images.githubusercontent.com/16235526/59819735-c781f280-935b-11e9-806d-5a542c4950cb.png)

有个问题想要请教一下，我做快速风格迁移的时候发现转换后的图片边框总是会乱掉，请问你有遇到过吗，怎么回事呢？

**Comment by LouieYang** [comment]
这是因为卷积的边界padding不连续导致的，我们实验中用的解决办法是在图片外padding（比如说20长度的0），然后forwarding之后再remove这个padding。

另一种脑补的方法是设置conv2d的参数padding=same，应该也是可行的。

@FantasyJXF

**Comment by FantasyJXF** [comment]
从代码里面可以看到你们在前向推理时加了boarder，在训练过程中所有的卷积都是SAME Padding.
```
img = np.array(load_image(args.content, 1024), dtype=np.float32)
border = np.ceil(np.shape(img)[0]/20/4).astype(int) * 5
container = [imresize(img, (np.shape(img)[0] + 2 * border, np.shape(img)[1] + 2 * border, 3))]
container[0][border : np.shape(img)[0] + border, border : np.shape(img)[1] + border, :] = img
container = np.repeat(container, train_batch_size, 0)
```

但是比较奇怪的是，并不是所有模型都出现这个情况。

![sketch_stylized](https://user-images.githubusercontent.com/16235526/59828064-c1484200-936c-11e9-87b8-d9b28547e16e.jpg)
![wave_stylized](https://user-images.githubusercontent.com/16235526/59828077-c86f5000-936c-11e9-88ec-aad7b7a84a49.jpg)

并且还比较容易出现在右侧，可能是跟原始的风格图片以及待转化的图片共同作用吧。
猜测是转置卷积上采样的时候哪里出了问题，随意猜想。
